spotify/app.py

```diff
@@ -10,215 +10,117 @@
 
 # TODO:
-# > Add "is_logged_in" check to block routes if a user isn't logged in yet.
 # > Make imports smaller at end of project by only importing dependencies.
 
 big_data = pd.read_csv("../spotify/data/data.csv").drop(columns=["explicit", 
                                                                  "year", 
                                                                  "release_date"])
 
+app = Flask(__name__)
+app.secret_key=secrets.token_bytes(16)
 
-def create_app():
-
-    app = Flask(__name__)
-    app.secret_key=secrets.token_bytes(16)
-
-    # Home
-    @app.route("/")
-    def index():
-        # Pulls login status to pass into renderer.
-        session["token_info"], authorized = get_token(session)
-        
-        # Keep the user on this page if not authorized otherwise direct to app.
-        if not authorized:
-            return render_template("index.html")
-        else:
-            return redirect(url_for("main_app"))
-
-
-    # Redirects users to login with spotify.
-    @app.route("/spotify_login")
-    def authorization():
-        sp = create_spotify_oauth()
-        auth_url = sp.get_authorize_url()
-
-        return redirect(auth_url)
-
-
-    # Welcomes users back from spotify login.
-    @app.route("/callback")
-    def callback():
-        sp = create_spotify_oauth()
-        session.clear()
-        code = request.args.get('code')
-        token_info = sp.get_access_token(code, check_cache=False)
-        
-        session["token_info"] = token_info
-
+# Home
+@app.route("/")
+def index():
+    # Pulls login status to pass into renderer.
+    session["token_info"], authorized = get_token(session)
+    
+    # Keep the user on this page if not authorized otherwise direct to app.
+    if not authorized:
+        return render_template("index.html")
+    else:
         return redirect(url_for("main_app"))
 
 
-    @app.route("/app", methods=["GET", "POST"])
-    def main_app():
-        sp = get_sp(session)
-
-        # Login check/redirect.
-        session["token_info"], authorized = get_token(session)
-        if not authorized:
-            return redirect(url_for("index"))
-
-        # The only POST request possible to this page is if the main app
-        # "Generate Predictions" button is clicked.
-        if request.method == 'GET':     
-            return render_template('main_app.html', sp=sp, songs=[])
-
-        # Pull user top tracks from spotify API.
-        user_top_tracks = sp.current_user_top_tracks(limit=10, 
-                                                     time_range="short_term"
-                                                    )["items"]
-        # Raise error if no recent top tracks.
-        if not user_top_tracks:
-            return render_template("error.html", type='Empty Spotify')
+# Redirects users to login with spotify.
+@app.route("/spotify_login")
+def authorization():
+    sp = create_spotify_oauth()
+    auth_url = sp.get_authorize_url()
```

Comment:
What happens when user doesn't have an acount?

Reply:
Users are redirected to verify with spotify and if they don't have a spotify account they have to make one.